src/execution_plan/optimizations/seek_by_id.c

```diff
@@ -128,16 +128,30 @@ static void _UseIdOptimization(ExecutionPlan *plan, OpBase *scan_op) {
 			}
 
 			_setupIdRange(rel, id, reverse, &minId, &maxId, &inclusiveMin, &inclusiveMax);
-			opNodeByIdSeek = NewNodeByIdSeekOp(scan_op->plan, node, minId, maxId,
-											   inclusiveMin, inclusiveMax);
-
-			// Managed to reduce!
-			ExecutionPlan_ReplaceOp(plan, scan_op, opNodeByIdSeek);
-			ExecutionPlan_RemoveOp(plan, (OpBase *)filter);
+			/* Don't replace label scan, but set it to have range query.
+			 * Issue 818 https://github.com/RedisGraph/RedisGraph/issues/818
+			 * This optimization caused a range query over the entire range of ids in the graph
+			 * regardless to the label. */
+			if(scan_op->type == OPType_NODE_BY_LABEL_SCAN) {
```

Comment:
I'd prefer if this logic were in `op_node_by_label_scan.c`, so that file gives some indication of how non-default values can be set.

Reply:
done